### PR TITLE
Add/Update AWS Audit Manager > NIST Cybersecurity Framework version 1.1 (2023-03-20). Closes #587

### LINKED ISSUE #587: Add/Update AWS Audit Manager > NIST Cybersecurity Framework version 1.1  (2023-03-20)
**Is your feature request related to a problem? Please describe.**

**New Additions**
[Audit Manager _ NIST-CSF(20MAR23) - nist_csf_missing_controls.pdf](https://github.com/turbot/steampipe-mod-aws-compliance/files/11017166/Audit.Manager._.NIST-CSF.20MAR23.-.nist_csf_missing_controls.pdf)

**Controls to remove**

Benchmark | Control(s)
-- | --
de_ae_1 | vpc_security_group_restrict_ingress_common_ports_all
de_cm_2 | config_enabled_all_regions
  | log_metric_filter_bucket_policy
  | log_metric_filter_cloudtrail_configuration
  | log_metric_filter_config_configuration
  | log_metric_filter_console_authentication_failure
  | log_metric_filter_console_login_mfa
  | log_metric_filter_disable_or_delete_cmk
  | log_metric_filter_iam_policy
  | log_metric_filter_network_acl
  | log_metric_filter_network_gateway
  | log_metric_filter_root_login
  | log_metric_filter_route_table
  | log_metric_filter_security_group
  | log_metric_filter_unauthorized_api
  | log_metric_filter_vpc
id_am_1 | config_enabled_all_regions
id_be_5 | elb_application_lb_deletion_protection_enabled
  | config_enabled_all_regions
  | log_metric_filter_bucket_policy
  | log_metric_filter_cloudtrail_configuration
  | log_metric_filter_config_configuration
  | log_metric_filter_console_authentication_failure
  | log_metric_filter_console_login_mfa
  | log_metric_filter_disable_or_delete_cmk
  | log_metric_filter_iam_policy
  | log_metric_filter_network_acl
  | log_metric_filter_network_gateway
  | log_metric_filter_root_login
  | log_metric_filter_route_table
  | log_metric_filter_security_group
  | log_metric_filter_unauthorized_api
  | log_metric_filter_vpc
id_sc_4 | config_enabled_all_regions
  | log_metric_filter_bucket_policy
  | log_metric_filter_cloudtrail_configuration
  | log_metric_filter_config_configuration
  | log_metric_filter_console_authentication_failure
  | log_metric_filter_console_login_mfa
  | log_metric_filter_disable_or_delete_cmk
  | log_metric_filter_iam_policy
  | log_metric_filter_network_acl
  | log_metric_filter_network_gateway
  | log_metric_filter_root_login
  | log_metric_filter_route_table
  | log_metric_filter_security_group
  | log_metric_filter_unauthorized_api
  | log_metric_filter_vpc
pr_ac_1 | iam_account_password_policy_strong_min_reuse_24
pr_ac_3 | s3_public_access_block_account
  | vpc_security_group_restrict_ingress_common_ports_all
pr_ac_5 | s3_public_access_block_account
  | vpc_security_group_restrict_ingress_common_ports_all
  | vpc_security_group_restrict_ingress_tcp_udp_all
pr_ds_1 | efs_file_system_encrypt_data_at_rest
  | sns_topic_encrypted_at_rest
pr_ds_3 | vpc_eip_associated
  | vpc_security_group_associated_to_eni
pr_ds_4 | elb_application_lb_deletion_protection_enabled
pr_ds_5 | s3_public_access_block_account
pr_ds_7 | cloudtrail_trail_validation_enabled
  | elb_application_lb_deletion_protection_enabled
  | vpc_security_group_restrict_ingress_ssh_all
pr_ip_12 | config_enabled_all_regions
pr_ip_3 | elb_application_lb_deletion_protection_enabled
pr_ip_8 | s3_public_access_block_account
pr_ip_9 | elb_application_lb_deletion_protection_enabled
pr_pt_3 | s3_public_access_block_account
pr_pt_4 | vpc_security_group_restrict_ingress_common_ports_all
pr_pt_5 | elb_application_lb_deletion_protection_enabled
rc_rp_1 | elb_application_lb_deletion_protection_enabled
rs_rp_1 | elb_application_lb_deletion_protection_enabled

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: nist_csf/function_de.sp [hunk 1]
```diff
@@ -28,19 +28,45 @@ benchmark "nist_csf_de_ae_1" {
...
+    control.rds_db_instance_logging_enabled,
     control.redshift_cluster_encryption_logging_enabled,
+    control.redshift_cluster_audit_logging_enabled,
     control.s3_bucket_logging_enabled,
+    control.s3_bucket_event_notifications_enabled,
     control.vpc_default_security_group_restricts_all_traffic,
     control.vpc_flow_logs_enabled,
+    control.vpc_network_acl_remote_administration,
+    control.vpc_route_table_restrict_public_access_to_igw,
+    control.vpc_security_group_allows_ingress_authorized_ports,
```
Comment (madhushreeray30) [inline nist_csf/function_de.sp:62]:
```suggestion
    control.vpc_security_group_restrict_ingress_tcp_udp_all,
```

File: nist_csf/function_de.sp [hunk 2]
```diff
@@ -66,10 +93,21 @@ benchmark "nist_csf_de_ae_3" {
...
+    control.opensearch_domain_logs_to_cloudwatch,
+    control.rds_db_instance_logging_enabled,
+    control.redshift_cluster_audit_logging_enabled,
     control.redshift_cluster_encryption_logging_enabled,
     control.s3_bucket_logging_enabled,
-    control.vpc_flow_logs_enabled
+    control.vpc_flow_logs_enabled,
+    control.waf_web_acl_logging_enabled,
+    control.wafv2_web_acl_logging_enabled
   ]
```
Comment (madhushreeray30) [inline nist_csf/function_de.sp:112]: These controls are not sorted, kindly sort them.

File: nist_csf/function_de.sp [hunk 3]
```diff
@@ -66,10 +93,21 @@ benchmark "nist_csf_de_ae_3" {
...
+    control.opensearch_domain_audit_logging_enabled,
+    control.opensearch_domain_logs_to_cloudwatch,
+    control.rds_db_instance_logging_enabled,
+    control.redshift_cluster_audit_logging_enabled,
     control.redshift_cluster_encryption_logging_enabled,
     control.s3_bucket_logging_enabled,
-    control.vpc_flow_logs_enabled
+    control.vpc_flow_logs_enabled,
+    control.waf_web_acl_logging_enabled,
+    control.wafv2_web_acl_logging_enabled
```
Comment (madhushreeray30) [inline nist_csf/function_de.sp:110]: These controls are not sorted, kindly sort them.

File: nist_csf/function_de.sp [hunk 4]
```diff
@@ -133,6 +171,7 @@ benchmark "nist_csf_de_cm_1" {
     control.elb_application_classic_lb_logging_enabled,
     control.guardduty_enabled,
     control.s3_bucket_logging_enabled,
+    control.s3_bucket_event_notifications_enabled,
```
Comment (madhushreeray30) [inline nist_csf/function_de.sp:174]: Sorting is incorrect.

File: nist_csf/function_de.sp [hunk 5]
```diff
@@ -165,7 +210,8 @@ benchmark "nist_csf_de_cm_2" {
     control.log_metric_filter_unauthorized_api,
     control.log_metric_filter_vpc,
     control.rds_db_instance_and_cluster_enhanced_monitoring_enabled,
-    control.securityhub_enabled
+    control.securityhub_enabled,
+    control.vpc_flow_logs_enabled,
```
Comment (madhushreeray30) [inline nist_csf/function_de.sp:214]:
```suggestion
    control.vpc_flow_logs_enabled
```

File: nist_csf/function_id.sp [hunk 6]
```diff
@@ -43,7 +43,10 @@ benchmark "nist_csf_id_am_2" {
 
   children = [
     control.ec2_instance_ssm_managed,
-    control.ssm_managed_instance_compliance_association_compliant
+    control.ssm_managed_instance_compliance_association_compliant,
+    control.ec2_stopped_instance_30_days,
+    control.vpc_eip_associated,
+    control.vpc_network_acl_unused
```
Comment (madhushreeray30) [inline nist_csf/function_id.sp:49]: Sorting is incorrect.

File: nist_csf/function_id.sp [hunk 7]
```diff
@@ -139,9 +163,36 @@ benchmark "nist_csf_id_ra_1" {
...
+    control.log_metric_filter_root_login,
+    control.log_metric_filter_route_table,
+    control.log_metric_filter_security_group,
+    control.log_metric_filter_unauthorized_api,
+    control.log_metric_filter_vpc,
+    control.rds_db_instance_and_cluster_enhanced_monitoring_enabled,
     control.securityhub_enabled,
-    control.ssm_managed_instance_compliance_patch_compliant
+    control.ssm_managed_instance_compliance_patch_compliant,
+    control.vpc_flow_logs_enabled,
```
Comment (madhushreeray30) [inline nist_csf/function_id.sp:195]:
```suggestion
    control.vpc_flow_logs_enabled
```

File: nist_csf/function_pr.sp [hunk 8]
```diff
@@ -186,17 +281,30 @@ benchmark "nist_csf_pr_ds_1" {
...
     control.rds_db_instance_encryption_at_rest_enabled,
+    control.redshift_cluster_encryption_logging_enabled,
+    control.redshift_cluster_kms_enabled,
     control.s3_bucket_default_encryption_enabled,
     control.s3_bucket_object_lock_enabled,
+    control.s3_bucket_default_encryption_enabled_kms,
     control.sagemaker_endpoint_configuration_encryption_at_rest_enabled,
     control.sagemaker_notebook_instance_encryption_at_rest_enabled,
-    control.sns_topic_encrypted_at_rest
+    control.secretsmanager_secret_encrypted_with_kms_cmk
```
Comment (madhushreeray30) [inline nist_csf/function_pr.sp:307]: Sorting is incorrect.

File: nist_csf/function_pr.sp [hunk 9]
```diff
@@ -240,13 +355,34 @@ benchmark "nist_csf_pr_ds_4" {
...
+    control.rds_db_instance_in_backup_plan,
+    control.rds_db_instance_deletion_protection_enabled,
     control.rds_db_instance_multiple_az_enabled,
+    control.redshift_cluster_automatic_snapshots_min_7_days,
+    control.redshift_cluster_maintenance_settings_check,
     control.s3_bucket_cross_region_replication_enabled,
     control.s3_bucket_versioning_enabled,
+    control.s3_bucket_lifecycle_policy_enabled,
+    control.s3_bucket_versioning_and_lifecycle_policy_enabled,
     control.vpc_vpn_tunnel_up
```
Comment (madhushreeray30) [inline nist_csf/function_pr.sp:386]: Sorting is incorrect.

File: nist_csf/function_pr.sp [hunk 10]
```diff
@@ -300,13 +446,13 @@ benchmark "nist_csf_pr_ds_7" {
...
-    control.cloudtrail_trail_validation_enabled,
-    control.ebs_attached_volume_delete_on_termination_enabled,
+    control.cloudtrail_security_trail_enabled,
     control.ec2_instance_ssm_managed,
     control.ec2_stopped_instance_30_days,
+    control.ebs_volume_unused,
     control.elb_application_lb_deletion_protection_enabled,
     control.ssm_managed_instance_compliance_association_compliant,
-    control.vpc_security_group_restrict_ingress_ssh_all
+    control.vpc_security_group_restrict_ingress_common_ports_all
```
Comment (madhushreeray30) [inline nist_csf/function_pr.sp:455]: Sorting is incorrect.

File: nist_csf/function_pr.sp [hunk 11]
```diff
@@ -348,10 +494,39 @@ benchmark "nist_csf_pr_ip_1" {
...
+    control.rds_db_instance_no_default_admin_name,
+    control.redshift_cluster_maintenance_settings_check,
+    control.redshift_cluster_no_default_admin_name,
+    control.redshift_cluster_no_default_database_name,
+    control.s3_bucket_policy_restricts_cross_account_permission_changes,
+    control.ssm_managed_instance_compliance_association_compliant,
+    control.waf_rule_condition_attached,
+    control.waf_rule_group_rule_attached,
+    control.waf_web_acl_resource_associated,
+    control.waf_regional_web_acl_rule_attached
```
Comment (madhushreeray30) [inline nist_csf/function_pr.sp:529]: Sorting is incorrect.

File: nist_csf/function_pr.sp [hunk 12]
```diff
@@ -388,11 +565,21 @@ benchmark "nist_csf_pr_ip_4" {
...
     control.elasticache_redis_cluster_automatic_backup_retention_15_days,
     control.rds_db_instance_backup_enabled,
+    control.rds_db_instance_in_backup_plan,
+    control.redshift_cluster_automatic_snapshots_min_7_days,
+    control.redshift_cluster_maintenance_settings_check,
     control.s3_bucket_cross_region_replication_enabled,
-    control.s3_bucket_versioning_enabled
+    control.s3_bucket_versioning_enabled,
+    control.s3_bucket_lifecycle_policy_enabled,
+    control.s3_bucket_versioning_and_lifecycle_policy_enabled
```
Comment (madhushreeray30) [inline nist_csf/function_pr.sp:582]: Sorting is incorrect.

File: nist_csf/function_pr.sp [hunk 13]
```diff
@@ -566,14 +793,41 @@ benchmark "nist_csf_pr_pt_4" {
...
+    control.vpc_igw_attached_to_authorized_vpc,
+    control.vpc_network_acl_remote_administration,
+    control.vpc_route_table_restrict_public_access_to_igw,
     control.vpc_security_group_restrict_ingress_common_ports_all,
     control.vpc_security_group_restrict_ingress_ssh_all,
-    control.vpc_security_group_restrict_ingress_tcp_udp_all
+    control.vpc_security_group_restrict_ingress_tcp_udp_all,
+    control.waf_regional_rule_condition_attached,
+    control.waf_regional_rule_group_rule_attached,
+    control.waf_web_acl_rule_attached,
```
Comment (madhushreeray30) [inline nist_csf/function_pr.sp:830]:
```suggestion
    control.waf_web_acl_rule_attached
```

File: nist_csf/function_rs.sp [hunk 14]
```diff
@@ -75,6 +75,7 @@ benchmark "nist_csf_rs_rp_1" {
   description = "Response plan is executed during or after an incident."
 
   children = [
+    control.backup_recovery_point_min_retention_35_days,
     control.backup_plan_min_retention_35_days,
     control.backup_recovery_point_encryption_enabled,
     control.backup_recovery_point_manual_deletion_disabled,
```
Comment (madhushreeray30) [inline nist_csf/function_rs.sp:81]: Sorting is incorrect.

**Review by madhushreeray30 — CHANGES_REQUESTED** [review]
@Priyanka-Chatterjee-2000 Can you please have a look at the comments! Thanks.

**Review by madhushreeray30 — APPROVED** [review]

File: conformance_pack/apigateway.sp [hunk 15]
```diff
@@ -63,10 +63,21 @@ control "apigateway_rest_api_stage_use_ssl_certificate" {
     hipaa_final_omnibus_security_rule_2013 = "true"
     nist_800_171_rev_2                     = "true"
     nist_800_53_rev_5                      = "true"
+    nist_csf                               = "true"
     rbi_cyber_security                     = "true"
   })
 }
 
+control "apigateway_rest_api_stage_xray_tracing_enabled" {
```
Comment (khushboo9024) [inline conformance_pack/apigateway.sp:71]: This is a duplicate control please check and remove.

File: conformance_pack/autoscaling.sp [hunk 16]
```diff
@@ -63,6 +65,36 @@ control "autoscaling_group_no_suspended_process" {
   })
 }
 
+control "autoscaling_group_multiple_az_configured" {
+  title       = "Amazon EC2 Auto Scaling group should cover multiple Availability Zones"
```
Comment (khushboo9024) [inline conformance_pack/autoscaling.sp:69]:
```suggestion
  title       = "EC2 auto scaling group should cover multiple availability zones"
```

File: conformance_pack/autoscaling.sp [hunk 17]
```diff
@@ -63,6 +65,36 @@ control "autoscaling_group_no_suspended_process" {
...
+  description = "This control checks whether an Auto Scaling group spans multiple Availability Zones. The control fails if an Auto Scaling group does not span multiple availability zones."
```
Comment (khushboo9024) [inline conformance_pack/autoscaling.sp:70]:
```suggestion
  description = "This control checks whether an auto scaling group spans multiple availability zones. The control fails if an auto scaling group does not span multiple availability zones."
```

File: conformance_pack/autoscaling.sp [hunk 18]
```diff
@@ -63,6 +65,36 @@ control "autoscaling_group_no_suspended_process" {
...
+  query       = query.autoscaling_group_multiple_az_configured
+
+  tags = merge(local.conformance_pack_autoscaling_common_tags, {
+    nist_csf = "true"
+  })
+}
+
+control "autoscaling_use_multiple_instance_types_in_multiple_az" {
+  title       = "Auto Scaling groups should use multiple instance types in multiple Availability Zones"
```
Comment (khushboo9024) [inline conformance_pack/autoscaling.sp:79]:
```suggestion
  title       = "EC2 auto scaling groups should use multiple instance types in multiple availability zones"
```

File: conformance_pack/autoscaling.sp [hunk 19]
```diff
@@ -63,6 +65,36 @@ control "autoscaling_group_no_suspended_process" {
...
+  description = "This control checks whether an Amazon EC2 Auto Scaling group uses multiple instance types. The control fails if the Auto Scaling group has only one instance type defined."
+  query       = query.autoscaling_use_multiple_instance_types_in_multiple_az
+
+  tags = merge(local.conformance_pack_autoscaling_common_tags, {
+    nist_csf = "true"
+  })
+}
+
+control "autoscaling_launch_config_hop_limit" {
+  title       = "Auto Scaling group launch configuration should not have metadata response hop limit greater than 1"
```
Comment (khushboo9024) [inline conformance_pack/autoscaling.sp:89]:
```suggestion
  title       = "EC2 auto scaling group launch configuration should not have metadata response hop limit greater than 1"
```

File: conformance_pack/cloudtrail.sp [hunk 20]
```diff
@@ -184,7 +187,18 @@ control "cloudtrail_bucket_not_public" {
...
 
   tags = merge(local.conformance_pack_cloudtrail_common_tags, {
-    gdpr = "true"
+    gdpr     = "true"
+    nist_csf = "true"
+  })
+}
+
+control "cloudtrail_multi_region_read_write_enabled" {
+  title       = "Ensure CloudTrail is enabled in all regions"
```
Comment (khushboo9024) [inline conformance_pack/cloudtrail.sp:196]:
```suggestion
  title       = "CloudTrail trails should be enabled in all regions"
```

File: conformance_pack/codebuild.sp [hunk 21]
```diff
@@ -102,8 +104,19 @@ control "codebuild_project_artifact_encryption_enabled" {
 
   tags = merge(local.conformance_pack_codebuild_common_tags, {
     cis_controls_v8_ig1 = "true"
-    gxp_21_cfr_part_11  = "true"
-    gxp_eu_annex_11     = "true"
+    nist_csf            = "true"
+  })
+}
+
+control "codebuild_project_s3_logs_encryption_enabled" {
```
Comment (khushboo9024) [inline conformance_pack/codebuild.sp:111]: This is a duplicate control, please remove.

File: conformance_pack/ec2.sp [hunk 22]
```diff
@@ -40,6 +42,26 @@ control "ec2_instance_detailed_monitoring_enabled" {
...
+  description = "This control checks whether an EC2 instance uses multiple Elastic Network Interfaces (ENIs) or Elastic Fabric Adapters (EFAs). This control passes if a single network adapter is used. The control includes an optional parameter list to identify the allowed ENIs."
+  query       = query.ec2_instance_not_use_multiple_enis
+
+  tags = merge(local.conformance_pack_ec2_common_tags, {
+    nist_csf = "true"
+  })
+}
+
+control "ec2_instance_no_amazon_key_pair" {
+  title       = "27 Running EC2 Instances should not use key pairs"
```
Comment (khushboo9024) [inline conformance_pack/ec2.sp:56]:
```suggestion
  title       = "EC2 instances should not use key pairs in running state"
```

File: conformance_pack/ec2.sp [hunk 23]
```diff
@@ -234,6 +259,17 @@ control "ec2_instance_no_launch_wizard_security_group" {
...
 }
 
+control "ec2_instance_virtualization_type_no_paravirtual" {
+  title       = "Paravirtual EC2 instance types should not be used"
+  description = "This control checks whether the virtualization type of an EC2 instance is paravirtual. The control fails if the virtualizationType of the EC2 instance is set to paravirtual."
+  query       = query.ec2_instance_virtualization_type_no_paravirtual
+
+  tags = merge(local.conformance_pack_ec2_common_tags, {
+    nist_csf     = "true"
+    other_checks = "true"
```
Comment (khushboo9024) [inline conformance_pack/ec2.sp:269]:
```suggestion
    nist_csf     = "true"
```

File: conformance_pack/ecs.sp [hunk 24]
```diff
@@ -72,6 +73,66 @@ control "ecs_task_definition_logging_enabled" {
   })
 }
 
+control "ecs_cluster_container_insights_enabled" {
+  title       = "ECS clusters should have Container Insights enabled"
```
Comment (khushboo9024) [inline conformance_pack/ecs.sp:77]:
```suggestion
  title       = "ECS clusters should have container insights enabled"
```

File: conformance_pack/ecs.sp [hunk 25]
```diff
@@ -72,6 +73,66 @@ control "ecs_task_definition_logging_enabled" {
...
+  description = "This control checks if Amazon ECS task definitions are configured to share a host’s process namespace with its containers. The control fails if the task definition shares the host's process namespace with the containers running on it."
+  query       = query.ecs_task_definition_no_host_pid_mode
+
+  tags = merge(local.conformance_pack_ecs_common_tags, {
+    nist_csf = "true"
+  })
+}
+
+control "ecs_service_fargate_using_latest_platform_version" {
+  title       = "Fargate services should run on the latest Fargate platform version"
```
Comment (khushboo9024) [inline conformance_pack/ecs.sp:127]:
```suggestion
  title       = "ECS fargate services should run on the latest fargate platform version"
```

File: conformance_pack/elb.sp [hunk 26]
```diff
@@ -299,6 +300,25 @@ control "elb_classic_lb_desync_mitigation_mode" {
   })
 }
 
+control "elb_classic_lb_multiple_az_configured" {
+  title       = "Classic Load Balancers should span multiple Availability Zones"
```
Comment (khushboo9024) [inline conformance_pack/elb.sp:304]:
```suggestion
  title       = "ELB classic load balancers should span multiple availability zones"
```

File: conformance_pack/apigateway.sp (hunk 15 above)
Comment (Priyanka-Chatterjee-2000) [reply conformance_pack/apigateway.sp:71]: This happened while resolving merge conflict.

File: conformance_pack/elb.sp [hunk 27]
```diff
@@ -299,6 +300,25 @@ control "elb_classic_lb_desync_mitigation_mode" {
...
+  description = "This control checks whether a Classic Load Balancer has been configured to span multiple Availability Zones. The control fails if the Classic Load Balancer does not span multiple Availability Zones."
+  query       = query.elb_classic_lb_multiple_az_configured
+
+  tags = merge(local.conformance_pack_elb_common_tags, {
+    nist_csf = "true"
+  })
+}
+
+control "elb_application_gateway_network_lb_multiple_az_configured" {
+  title       = "Application, Network, and Gateway Load Balancers should span multiple Availability Zones"
```
Comment (khushboo9024) [inline conformance_pack/elb.sp:314]:
```suggestion
  title       = "ELB application, network, and gateway load balancers should span multiple availability zones"
```

File: conformance_pack/codebuild.sp (hunk 21 above)
Comment (Priyanka-Chatterjee-2000) [reply conformance_pack/codebuild.sp:111]: This happened while resolving merge conflict.

File: conformance_pack/ec2.sp (hunk 23 above)
Comment (Priyanka-Chatterjee-2000) [reply conformance_pack/ec2.sp:269]: This happened while resolving merge conflict.

File: conformance_pack/lambda.sp [hunk 28]
```diff
@@ -106,6 +106,26 @@ control "lambda_function_tracing_enabled" {
   })
 }
 
+control "lambda_function_multiple_az_configured" {
+  title       = "VPC Lambda functions should operate in more than one Availability Zone"
```
Comment (khushboo9024) [inline conformance_pack/lambda.sp:110]:
```suggestion
  title       = "Lambda functions should operate in more than one availability zone"
```

File: conformance_pack/opensearch.sp [hunk 29]
```diff
@@ -24,10 +25,35 @@ control "opensearch_domain_fine_grained_access_enabled" {
   query       = query.opensearch_domain_fine_grained_access_enabled
 
   tags = merge(local.conformance_pack_opensearch_common_tags, {
+    nist_csf     = "true"
     pci_dss_v321 = "true"
   })
 }
 
+control "opensearch_domain_in_vpc" {
```
Comment (khushboo9024) [inline conformance_pack/opensearch.sp:33]: This is also a duplicate control.

File: conformance_pack/opensearch.sp [hunk 30]
```diff
@@ -24,10 +25,35 @@ control "opensearch_domain_fine_grained_access_enabled" {
...
+  description = "This control checks whether Amazon OpenSearch domains are in a VPC. It does not evaluate the VPC subnet routing configuration to determine public access."
+  query       = query.opensearch_domain_in_vpc
+
+  tags = merge(local.conformance_pack_opensearch_common_tags, {
+    hipaa_security_rule_2003 = "true"
+    nist_csf                 = "true"
+  })
+}
+
+control "opensearch_domain_node_to_node_encryption_enabled" {
```
Comment (khushboo9024) [inline conformance_pack/opensearch.sp:44]: This is a duplicate control.

File: conformance_pack/s3.sp [hunk 31]
```diff
@@ -348,7 +352,30 @@ control "s3_bucket_event_notifications_enabled" {
...
+  title       = "S3 access control lists (ACLs) should not be used to manage user access to buckets"
+  description = "This control checks whether Amazon S3 buckets provide user permissions via ACLs. The control fails if ACLs are configured for managing user access on S3 buckets."
+  query       = query.s3_bucket_acls_should_prohibit_user_access
+
+  tags = merge(local.conformance_pack_s3_common_tags, {
+    nist_csf = "true"
+  })
+}
+
+control "s3_bucket_policy_restrict_public_access" {
```
Comment (khushboo9024) [inline conformance_pack/s3.sp:370]: This is also a duplicate control.

File: conformance_pack/vpc.sp [hunk 32]
```diff
@@ -305,6 +307,17 @@ control "vpc_network_acl_unused" {
   tags = merge(local.conformance_pack_vpc_common_tags, {
     cis_controls_v8_ig1   = "true"
     cisa_cyber_essentials = "true"
+    nist_csf              = "true"
+  })
+}
+
+control "vpc_security_group_allows_ingress_authorized_ports" {
```
Comment (khushboo9024) [inline conformance_pack/vpc.sp:314]: Duplicate control.

File: conformance_pack/waf.sp [hunk 33]
```diff
@@ -52,20 +56,45 @@ control "waf_regional_rule_condition_attached" {
...
 }
 
+control "waf_regional_web_acl_rule_attached" {
+  title       = "WAF regional web ACL should have at least one rule or rule group attached"
+  description = "This control checks if a WAF regional Web ACL contains any WAF rules or rule groups. The rule is non compliant if there are no WAF rules or rule groups present within a Web ACL."
+  query       = query.waf_regional_web_acl_rule_attached
+
+  tags = merge(local.conformance_pack_waf_common_tags, {
+    audit_manager_pci_v321 = "true"
+    nist_csf               = "true"
```
Comment (khushboo9024) [inline conformance_pack/waf.sp:94]:
```suggestion
    nist_csf               = "true"
```

File: conformance_pack/waf.sp [hunk 34]
```diff
@@ -178,3 +207,79 @@ query "waf_web_acl_logging_enabled" {
       aws_waf_web_acl;
   EOQ
 }
+
+query "waf_regional_rule_condition_attached" {
```
Comment (khushboo9024) [inline conformance_pack/waf.sp:211]: Query is duplicate

File: conformance_pack/waf.sp [hunk 35]
```diff
@@ -178,3 +207,79 @@ query "waf_web_acl_logging_enabled" {
...
+        when predicates is null or jsonb_array_length(predicates) = 0 then title || ' has no attached conditions.'
+        else title || ' has attached conditions.'
+      end as reason
+      ${local.common_dimensions_sql}
+    from
+      aws_wafregional_rule;
+  EOQ
+}
+
+query "waf_web_acl_resource_associated" {
```
Comment (khushboo9024) [inline conformance_pack/waf.sp:229]: Query is duplicate

File: conformance_pack/waf.sp [hunk 36]
```diff
@@ -52,20 +56,44 @@ control "waf_regional_rule_condition_attached" {
...
 
+control "waf_regional_rule_group_rule_attached" {
+  title       = "WAF regional rule group should have at least one rule attached"
+  description = "This control checks if WAF regional rule groups contain any rules. The rule is non compliant if there are no rules present within a WAF regional rule group."
+  query       = query.waf_regional_rule_group_rule_attached
+
+  tags = merge(local.conformance_pack_waf_common_tags, {
+    audit_manager_pci_v321 = "true"
+    nist_csf               = "true"
+    well_architected       = "true"
```
Comment (khushboo9024) [inline conformance_pack/waf.sp:72]: Tags are wrong.

File: conformance_pack/opensearch.sp [hunk 37]
```diff
@@ -85,8 +92,8 @@ control "opensearch_domain_in_vpc" {
   query       = query.opensearch_domain_in_vpc
 
   tags = merge(local.conformance_pack_opensearch_common_tags, {
-    hipaa_final_omnibus_security_rule_2013 = "true"
-    hipaa_security_rule_2003               = "true"
+    hipaa_security_rule_2003 = "true"
+    nist_csf                 = "true"
```
Comment (khushboo9024) [inline conformance_pack/opensearch.sp:96]:
```suggestion
   hipaa_final_omnibus_security_rule_2013 = "true
    hipaa_security_rule_2003                         = "true"
    nist_csf                                                       = "true"
```

File: conformance_pack/opensearch.sp [hunk 38]
```diff
@@ -61,21 +66,23 @@ control "opensearch_domain_logs_to_cloudwatch" {
 
   tags = merge(local.conformance_pack_opensearch_common_tags, {
     gxp_21_cfr_part_11 = "true"
+    gxp_eu_annex_11    = "true"
```
Comment (khushboo9024) [inline conformance_pack/opensearch.sp:69]:
```suggestion
```

File: conformance_pack/rds.sp [hunk 39]
```diff
@@ -223,8 +225,10 @@ control "rds_db_instance_deletion_protection_enabled" {
     ffiec                                  = "true"
     gxp_21_cfr_part_11                     = "true"
     hipaa_final_omnibus_security_rule_2013 = "true"
+    nist_800_171_rev_2                     = "true"
```
Comment (khushboo9024) [inline conformance_pack/rds.sp:228]:
```suggestion
```

File: conformance_pack/codebuild.sp [hunk 40]
```diff
@@ -102,19 +105,19 @@ control "codebuild_project_artifact_encryption_enabled" {
 
   tags = merge(local.conformance_pack_codebuild_common_tags, {
     cis_controls_v8_ig1 = "true"
-    gxp_21_cfr_part_11  = "true"
-    gxp_eu_annex_11     = "true"
```
Comment (khushboo9024) [inline conformance_pack/codebuild.sp:106]: Tags are wrong.

File: conformance_pack/ec2.sp [hunk 41]
```diff
@@ -17,8 +17,10 @@ control "ec2_ebs_default_encryption_enabled" {
     gxp_eu_annex_11                        = "true"
     hipaa_final_omnibus_security_rule_2013 = "true"
     hipaa_security_rule_2003               = "true"
+    nist_800_171_rev_2                     = "true"
```
Comment (khushboo9024) [inline conformance_pack/ec2.sp:20]:
```suggestion
```

File: conformance_pack/s3.sp [hunk 42]
```diff
@@ -234,12 +234,9 @@ control "s3_public_access_block_bucket_account" {
   query       = query.s3_public_access_block_bucket_account
 
   tags = merge(local.conformance_pack_s3_common_tags, {
-    fedramp_low_rev_4      = "true"
     fedramp_moderate_rev_4 = "true"
     ffiec                  = "true"
-    nist_800_171_rev_2     = "true"
     nist_800_53_rev_4      = "true"
-    nist_csf               = "true"
```
Comment (misraved) [inline conformance_pack/s3.sp:242]: Why were all the tags removed?

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!

File: conformance_pack/ec2.sp [hunk 43]
```diff
@@ -272,6 +306,44 @@ query "ec2_instance_detailed_monitoring_enabled" {
   EOQ
 }
 
+query "ec2_instance_not_use_multiple_enis" {
+  sql = <<-EOQ
+    select
+      arn as resource,
+      case
+        when jsonb_array_length(network_interfaces) = 1 then 'ok'
+        else 'alarm'
```
[the remark this answers is not in the document]
Comment (khushboo9024) [reply conformance_pack/ec2.sp:315]: This cannot be a case as every Amazon EC2 instance has a primary ENI on eth0 attached.

File: conformance_pack/iam.sp [hunk 44]
```diff
@@ -190,6 +191,7 @@ control "iam_user_mfa_enabled" {
 
   tags = merge(local.conformance_pack_iam_common_tags, {
     audit_manager_control_tower            = "true"
+    cis_controls_v8_ig1                    = "true"
```
[the remark this answers is not in the document]
Comment (khushboo9024) [reply conformance_pack/iam.sp:194]: yes we missed to add it previously.

File: conformance_pack/s3.sp (hunk 42 above)
Comment (khushboo9024) [reply conformance_pack/s3.sp:242]: We are removing this as tagging was incorrect .

**Review by misraved — APPROVED** [review]

**Pull request merged**